docs/rfcs/1003-mandatory-human-handle.md

```diff
@@ -0,0 +1,205 @@
+- redacted: which is the regular one, minus human handle / device label fields
+  which have been set to None
+
+## The issue
+
+The human handle / device label fields are almost exclusively used by the GUI (the lower
+layers using the user id / device id).
+
+It is annoying for the GUI to have to deal with the fact those fields might be missing,
+and in such case to roll a custom logic to still have something to display.
+
+This RFC propose to move the custom logic with the libparsec lower layer, so that
+the GUI has the guarantee those fields are always present from it point of view.
+
+## The solution
+
+tl;dr:
+
+1 - server enforce non-redacted certificates have a human handle / device label
+2 - when deserializing device certificate with no device label, device name is used instead
+3 - when deserializing user certificate with no human handle:
+
+  - user id is used as human handle label
+  - `<uncaseify(user_id)>@redacted.invalid` is used as human handle email (see
+    [section 3](### 3) for what is `uncaseify`)
+
+4 - Real human handle are no longer allowed to use the `redacted.invalid` domain
+5 - the human handle doesn't carry the fact it has been generated, the GUI is expected
+  instead to determine this information from the context.
+
+### 1 - Human handle / device label mandatory for non-redacted certificate
+
+This is already what is done in practice, but we now have to make it a rule (just like
+a redacted certificate is not allowed to have those fields present).
+
+The change is minimal: just return a new error status server-side, change nothing in the
+client given this status should never occur (and if it occurs anyway it will be caught
+by the generic error status handler)
+
+Of course, this rule cannot be enforced when loading existing certificates given
+they could have been created with Parsec<1.13.
+Instead we rely on specific deserialization rules to handle missing human handle /
+device label field (see below).
+The big pro of dealing at deserialization is we need no specific code to deal with
+legacy certificates: they are simply handled as regular redacted certificates.
+
+### 2 - Device label for redacted certificate
+
+Device label is trivial to handle: `DeviceLabel` type is a superset of `DeviceName`,
+hence the device's device name (which is always present) can be provided as device label.
+
+In practice what we get:
+
+- Device certificate created with Parsec<1.13: `MyPC` (i.e. device name field)
+- Non-redacted certificate: `My PC` (i.e. device label field)
+- Redacted certificate: `e5904a3eccd8451b99ba09324104662e` (i.e. device name field, which is always a UUID)
+
+### 3 - Human handle for redacted certificate
+
+Just like for device label, human handle label is a superset of user id so the latter can
+be directly used.
+
+For the human handle email we have to be more cunning (especially given the email *is* the
+field used as identifier where label is only displayed to human with no other purpose
```

Comment:
It is not clear to me why (or where) "the email is the field used as identifier", isn't the user id always use as identifier? 

Which makes me think, do we really need to bother at all with the email's local-part? How about always using something like `redacted@redacted.invalid`?

Reply:
This is indeed interesting for the documentation (not sure if the user documentation though, or maybe dev/admin docs?)

Reply:
I would say this has it place in the regular: it explain a fundamental concept of Parsec that is useful for advanced end up (the kind of one that read documentation ^^)

typically we want to expose userid/deviceid/realmid in the GUI (in an advanced property section hidden by default) because it helps when working with server administrator